api/envoy/config/filter/http/csrf/v2/csrf.proto

```diff
@@ -15,9 +15,27 @@ import "gogoproto/gogo.proto";
 // [#protodoc-title: CSRF]
 // Cross-Site Request Forgery :ref:`configuration overview <config_http_filters_csrf>`.
 
+// Origin Matcher for CSRF policy configuration.
+message OriginMatcher {
+  // A string containing either an exact origin or regex pattern.
+  string origin = 1 [(validate.rules).string.min_bytes = 1];
```

Comment:
Any reason to not just use `StringMatcher` here?

Reply:
It looks like `StringMatcher` allows for prefix and suffix matching which is something we would probably want to avoid here. Since this is a whitelist of origins to allow I worry that allowing prefix or suffix matches would open up more potential for mistakes.

I realize that the same mistakes can occur with generous regex patterns and contemplated allowing regex at all for this reason. However, I do think there are reasons where regex might make sense (if you have a variable host name in some a dev environment).

What are your thoughts @mattklein123 ? I can see how if we support regex here it would be almost the same as supporting prefix/suffix.

Reply:
IMO I would just use StringMatcher if you are going to support regex, and maybe add a big warning do the docs with recommendations?